dev/src/reference.ts

```diff
@@ -3051,76 +3052,127 @@ export class AggregateQuery<T extends firestore.AggregateSpec>
+    const otherAggregateKeys = Object.keys(other._aggregates).sort();
+    if (!isPrimitiveArrayEqual(thisAggregateKeys, otherAggregateKeys)) {
+      return false;
     }
-    return false;
+
+    // TODO: Also compare the `AggregateField` instances from `this._aggregates`
+    // to their counterparts in `other._aggregates` once `AggregateField` gains
+    // an `isEqual()` method, which will happen once we support more than one
+    // type of `AggregateField` (currently, we only support "count").
+    return true;
   }
 }
 
+/**
+ * The results of executing an aggregation query.
+ */
 export class AggregateQuerySnapshot<T extends firestore.AggregateSpec>
   implements firestore.AggregateQuerySnapshot<T>
 {
+  /**
+   * @private
+   * @internal
+   *
+   * @param _query The query that was executed to produce this result.
+   * @param _readTime The time this snapshot was read.
+   * @param _data The results of the aggregations performed over the underlying
+   * query.
+   */
   constructor(
     private readonly _query: AggregateQuery<T>,
     private readonly _readTime: Timestamp,
     private readonly _data: firestore.AggregateSpecData<T>
   ) {}
 
-  get query(): firestore.AggregateQuery<T> {
+  /** The query that was executed to produce this result. */
+  get query(): AggregateQuery<T> {
     return this._query;
   }
 
-  get readTime(): firestore.Timestamp {
+  /** The time this snapshot was read. */
+  get readTime(): Timestamp {
     return this._readTime;
   }
 
+  /**
+   * Returns the results of the aggregations performed over the underlying
+   * query.
+   *
+   * The keys of the returned object will be the same as those of the
+   * `AggregateSpec` object specified to the aggregation method, and the
+   * values will be the corresponding aggregation result.
+   *
+   * @returns The results of the aggregations performed over the underlying
+   * query.
+   */
+  data(): firestore.AggregateSpecData<T> {
+    return this._data;
+  }
+
+  /**
+   * Compares this object with the given object for equality.
+   *
+   * Two `AggregateQuerySnapshot` instances are considered "equal" if they
+   * have the read time, the same data, and underlying queries that compare
+   * "equal" using `AggregateQuery.isEqual()`.
+   *
+   * @param other The object to compare to this object for equality.
+   * @return `true` if this object is "equal" to the given object, as
+   * defined above, or `false` otherwise.
+   */
   isEqual(other: firestore.AggregateQuerySnapshot<T>): boolean {
     if (this === other) {
       return true;
     }
-    if (other instanceof AggregateQuerySnapshot) {
-      if (!this._query.isEqual(other._query)) {
-        return false;
-      }
-
-      const thisData = this._data;
-      const thisDataKeys: string[] = Object.keys(thisData);
+    if (!(other instanceof AggregateQuerySnapshot)) {
+      return false;
+    }
+    if (!this.readTime.isEqual(other.readTime)) {
```

Comment:
The isEquals method for QuerySnapshot does not include timestamp. Are you sure this is correct behavior?